src/adventure/lineparsers.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <QString>
+
+class AbstractStatefulLineParser
```

Comment:
Maybe we need to combine these two? https://github.com/MUME/MMapper/blob/master/src/parser/Action.h